zxlive/editor_base_panel.py

```diff
@@ -313,7 +342,16 @@ def toolbar_select_node_edge(parent: EditorBasePanel) -> ToolbarSection:
     select.clicked.connect(lambda: parent._tool_clicked(ToolType.SELECT))
     vertex.clicked.connect(lambda: parent._tool_clicked(ToolType.VERTEX))
     edge.clicked.connect(lambda: parent._tool_clicked(ToolType.EDGE))
-    return ToolbarSection(select, vertex, edge, exclusive=True)
+    yield ToolbarSection(select, vertex, edge, exclusive=True)
+
+    snap = QToolButton(parent)
+    snap.setCheckable(True)
+    snap.setChecked(True)
+    snap.setText("Vertex-edge snap")
```

Comment:
Not necessary for this PR but it would be nice to have a sleek icon instead of the text button

Reply:
I added an icon which I think looks pretty neat :)
(courtesy of chatGPT with a dash of my own creative inspiration)